src/systems/systems/src/air_conditioning/acs_controller.rs

```diff
@@ -0,0 +1,1808 @@
+    }
+
+    fn calculate_duct_temp_lower_limit(&self) -> ThermodynamicTemperature {
+        if self.zone_measured_temperature > ThermodynamicTemperature::new::<degree_celsius>(28.) {
+            ThermodynamicTemperature::new::<kelvin>(Self::LOWER_DUCT_TEMP_LIMIT_LOW)
+        } else if self.zone_measured_temperature
+            < ThermodynamicTemperature::new::<degree_celsius>(26.)
+        {
+            ThermodynamicTemperature::new::<kelvin>(Self::LOWER_DUCT_TEMP_LIMIT_HIGH)
+        } else {
+            let interpolation =
+                (Self::LOWER_DUCT_TEMP_LIMIT_LOW - Self::LOWER_DUCT_TEMP_LIMIT_HIGH) / (28. - 26.)
+                    * (self.zone_measured_temperature.get::<kelvin>() - 299.15)
+                    + Self::LOWER_DUCT_TEMP_LIMIT_HIGH;
+            ThermodynamicTemperature::new::<kelvin>(interpolation)
+        }
+    }
+
+    fn zone_id(&self) -> &str {
+        self.zone_id
+    }
+}
+
+impl DuctTemperature for ZoneController {
+    fn duct_demand_temperature(&self) -> HashMap<&'static str, ThermodynamicTemperature> {
+        let mut demand_temperature: HashMap<&str, ThermodynamicTemperature> = HashMap::new();
+        demand_temperature.insert(self.zone_id, self.duct_demand_temperature);
+        demand_temperature
+    }
+}
+
+impl SimulationElement for ZoneController {
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        self.zone_measured_temperature = reader.read(&self.zone_temp_id);
+    }
+
+    fn write(&self, writer: &mut SimulatorWriter) {
+        // TODO: Replace this with actual duct temperature when mixer is modelled, not duct demand temperature
+        writer.write(&self.zone_duct_temp_id, self.duct_demand_temperature);
+    }
+}
+
+#[derive(Clone, Copy)]
+enum OvhdFlowSelector {
+    Lo = 80,
+    Norm = 100,
+    Hi = 120,
+}
+
+read_write_enum!(OvhdFlowSelector);
+
+impl From<f64> for OvhdFlowSelector {
+    fn from(value: f64) -> Self {
+        match value as u8 {
+            0 => OvhdFlowSelector::Lo,
+            1 => OvhdFlowSelector::Norm,
+            2 => OvhdFlowSelector::Hi,
+            _ => panic!("Overhead flow selector position not recognized."),
+        }
+    }
+}
+
+struct PackFlowController {
+    apu_bleed_valve_open_id: VariableIdentifier,
+    apu_rpm_id: VariableIdentifier,
+    crossbleed_id: VariableIdentifier,
+    ditching_id: VariableIdentifier,
+    engine_1_state_id: VariableIdentifier,
+    engine_2_state_id: VariableIdentifier,
+    eng_1_fire_id: VariableIdentifier,
+    eng_2_fire_id: VariableIdentifier,
+    ovhd_flow_selector_id: VariableIdentifier,
+
+    pack_flow_id: VariableIdentifier,
+
+    flow_demand: Ratio,
+    absolute_flow: MassRate,
+    fcv_1_open_allowed: bool,
+    fcv_2_open_allowed: bool,
+    should_open_fcv: [bool; 2],
+
+    apu_bleed_on: bool,
+    crossbleed_is_on: bool,
+    ditching_is_on: bool,
+    engine_1_in_start_mode: bool,
+    engine_2_in_start_mode: bool,
+    engine_1_on_fire: bool,
+    engine_2_on_fire: bool,
```

Comment:
**thought**: I'm not quite sure if the current logic accurately represents the engine being on fire.
**thought**: Maybe you care about the engine fire pb being released more than a fire indication?

Reply:
I changed this to pull from the overhead `EngineFirePushButtons` instead.